multispecies_whale_detection/front_end.py

```diff
@@ -181,7 +217,30 @@ def call(self, waveform, training=False):
     num_frequency_bins = magnitude.shape[-1]
     hz_per_bin = sample_rate / 2 / num_frequency_bins
 
-    db = amplitude_ratio_to_db(magnitude + self._config.log_stabilizer)
+    frequency_scaling = self._config.frequency_scaling
+    if not frequency_scaling:
+      db = amplitude_ratio_to_db(magnitude + self._config.log_stabilizer)
+    elif isinstance(frequency_scaling, MelScalingConfig):
+      mel_scale_power = tf.matmul(
+          tf.square(magnitude),
+          tf.signal.linear_to_mel_weight_matrix(
+              num_mel_bins=frequency_scaling.num_mel_bins,
+              num_spectrogram_bins=num_frequency_bins,
+              sample_rate=sample_rate,
+              lower_edge_hertz=frequency_scaling.lower_edge_hz,
+              upper_edge_hertz=(frequency_scaling.upper_edge_hz or
+                                float(sample_rate / 2)),
+          ),
+      )
+      db = power_ratio_to_db(mel_scale_power +
+                             tf.square(self._config.log_stabilizer))
```

Comment:
Yes, the motivation is consistent floor on the dB scale.

Punting to another PR, though, since I've noticed the _to_db functions are actually wrong (!), and this falls under that theme.

Probably changing to "stabilizer_db" with a docstring about that being effectively a floor would both help code readability and make use more intuitive.